backend/src/main/java/com/example/parkingspot/controller/EventController.java

```diff
@@ -17,30 +19,33 @@
 
 import com.example.parkingspot.entity.Event;
 import com.example.parkingspot.service.EventService;
+import com.example.parkingspot.service.UserService;
 
 @RestController
 @RequestMapping("/api/events")
 public class EventController {
 
   private EventService eventService;
+  private UserService userService;
 
-  public EventController(EventService eventService) {
+  public EventController(EventService eventService, UserService userService) {
     this.eventService = eventService;
+    this.userService = userService;
   }
 
   // @GetMapping
   // public List<Event> getAllEvents() {
   // return eventService.fetchAllEvents();
   // }
 
-  @GetMapping("/{id}")
-  public ResponseEntity<Event> getEventById(@PathVariable("id") Long eventId) {
-    Event foundEvent = eventService.fetchEventById(eventId);
+  @GetMapping()
+  public ResponseEntity <List<Event>> getEventById() {
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    String username = auth.getName();
+    var userId = userService.getUserId(username);
+    var foundEvent = eventService.fetchEventByUserId(userId);
```

Comment:
Nice use of SecurityContextHolder instead of glued solution with token and localstorage